### PR TITLE
Camera

### PR DESCRIPTION
**Pull request by pavelLyolya**

### CODE REVIEW

File: src/screens/Camera/Camera.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  state = {
+    takenShot: null,
+    isSavedNotificationShown: false,
+    orientation: ORIENTATION.PORTRAIT,
+    flashMode: RNCamera.Constants.FlashMode.off,
+    cameraType: RNCamera.Constants.Type.back,
+  }
+
+  componentDidMount() {
+    Orientation.getDeviceOrientation((orientation) => this.setState({ orientation }));
```
Comment (nataly-chobat) [inline src/screens/Camera/Camera.js]: You can call `onDeviceOrientationDidChange` instead of setting state

File: src/screens/Camera/Camera.js [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+  savePicture = async () => {
+    if (this.state.takenShot) {
+      await CameraRoll.saveToCameraRoll(this.state.takenShot);
+    }
+    await this.retakePicture();
+    this.showSavedNotification();
+  }
+
+  showSavedNotification = () => {
```
Comment (nataly-chobat) [inline src/screens/Camera/Camera.js]: Did you try `toggleSaveNotification` method which will set `true/false` based on previous state?

File: src/screens/Camera/Camera.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    this.setState({isSavedNotificationShown: true});
+  }
+
+  hideSavedNotification = () => {
+    this.setState({isSavedNotificationShown: false});
+  }
+
+  toggleFlash = () => {
+    if (this.state.flashMode === RNCamera.Constants.FlashMode.on) {
```
Comment (nataly-chobat) [inline src/screens/Camera/Camera.js]: Let's try to rewrite with `prevState`

File: src/screens/Camera/Camera.js [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+      this.setState({ flashMode: RNCamera.Constants.FlashMode.off });
+    } else {
+      this.setState({ flashMode: RNCamera.Constants.FlashMode.on });
+    }
+  }
+
+  toggleCamera = () => {
+    if (this.state.cameraType === RNCamera.Constants.Type.back) {
```
Comment (nataly-chobat) [inline src/screens/Camera/Camera.js]: Let's try to rewrite with `prevState`

File: src/screens/Camera/Camera.js [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    }
+    return (
+      <View style={styles.container}>
+        {
+          this.state.isSavedNotificationShown
+          && <SavedNotification hideFn={this.hideSavedNotification} />
+        }
+        <View style={styles.preview}>
+          <RNCamera
+            ref={ref => {
```
Comment (nataly-chobat) [inline src/screens/Camera/Camera.js]: Did you read about `createRef()` functionality? It is a new version for ref creation

Comment (pavelLyolya) [reply src/screens/Camera/Camera.js]: yep! in docs they give two ways of refs creation :)

**Pull request merged**